addons/website_sale/models/product_template.py

```diff
@@ -189,7 +190,7 @@ def _get_sales_prices(self, pricelist):
                 'price_reduce': price_reduce
             }
             base_price = None
-            price_list_contains_template = price_reduce != base_sales_prices[template.id]
+            price_list_contains_template = float_compare(price_reduce, base_sales_prices[template.id], precision_digits=2) != 0
```

Comment:
For price comparisons, you should use `compare_amounts` of the `res.currency` model.
You cannot arbitrarily fix the decimal precision to 2 like that, it varies between currencies.
```suggestion
            price_list_contains_template = pricelist.currency_id.compare_amounts(price_reduce, base_sales_prices[template.id]) != 0
```